flake.nix

```diff
@@ -2,193 +2,252 @@
   description = "TeslaMate Logger";
 
   inputs = {
-    nixpkgs = { url = "github:NixOS/nixpkgs/nixos-unstable"; };
-    flake-utils = { url = "github:numtide/flake-utils"; };
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    flake-parts.url = "github:hercules-ci/flake-parts";
+    flake-parts.inputs.nixpkgs-lib.follows = "nixpkgs";
     devenv.url = "github:cachix/devenv";
+    treefmt-nix.url = "github:numtide/treefmt-nix";
+    treefmt-nix.inputs.nixpkgs.follows = "nixpkgs";
   };
 
-  outputs = inputs@{ self, nixpkgs, flake-utils, devenv }:
-    (flake-utils.lib.eachDefaultSystem (system:
-      let
-        inherit (pkgs.lib) optional optionals;
-        pkgs = nixpkgs.legacyPackages.${system};
-
-        elixir = pkgs.beam.packages.erlang.elixir_1_16;
-        beamPackages = pkgs.beam.packagesWith pkgs.beam.interpreters.erlang;
-
-        src = ./.;
-        version = builtins.readFile ./VERSION;
-        pname = "teslamate";
-
-        mixFodDeps = beamPackages.fetchMixDeps {
-          TOP_SRC = src;
-          pname = "${pname}-mix-deps";
-          inherit src version;
-          hash = "sha256-CeGtWKLjZJsoHk+uSAIz8s46XyGJWa3ECm5rSRfqlrc=";
-          # hash = pkgs.lib.fakeHash;
-        };
-
-        nodejs = pkgs.nodejs;
-        nodePackages = pkgs.buildNpmPackage {
-          name = "teslamate";
-          src = ./assets;
-          npmDepsHash = "sha256-05AKPyms4WP8MHBqWMup8VXR3a1tv/f/7jT8c6EpWBw=";
-          # npmDepsHash = pkgs.lib.fakeHash;
-          dontNpmBuild = true;
-          inherit nodejs;
-
-          installPhase = ''
-            mkdir $out
-            cp -r node_modules $out
-            ln -s $out/node_modules/.bin $out/bin
-
-            rm $out/node_modules/phoenix
-            ln -s ${mixFodDeps}/phoenix $out/node_modules
-
-            rm $out/node_modules/phoenix_html
-            ln -s ${mixFodDeps}/phoenix_html $out/node_modules
-
-            rm $out/node_modules/phoenix_live_view
-            ln -s ${mixFodDeps}/phoenix_live_view $out/node_modules
-          '';
-        };
+  outputs =
+    inputs@{
+      self,
+      flake-parts,
+      devenv,
+      ...
+    }:
+    flake-parts.lib.mkFlake { inherit inputs; } {
+
+      systems = [
+        "x86_64-linux"
+        "aarch64-linux"
+        "x86_64-darwin"
+        "aarch64-darwin"
+      ];
+      # See ./nix/modules/*.nix for the modules that are imported here.
+      imports = [
+        inputs.devenv.flakeModule
+      ] ++ builtins.map (fn: ./nix/modules/${fn}) (builtins.attrNames (builtins.readDir ./nix/modules));
```

Comment:
It is tempting to be clever and automatically discover modules to load, however IMO it generally better to just list them all out explicitly (especially when we only have a small handful of modules). It makes it more clear what files are being imported and gives us more control if we want to do anything beyond a flat directory with nothing but *.nix flake modules.

Reply:
agree on this